src/scene/materials/standard-material.js

```diff
@@ -356,21 +348,13 @@ class StandardMaterial extends Material {
     }
 
     reset() {
-        for (let i = 0; i < _propsSerial.length; i++) {
-            const defVal = _propsSerialDefaultVal[i];
-            this[_propsSerial[i]] = defVal ? (defVal.clone ? defVal.clone() : defVal) : defVal;
-        }
-        for (let i = 0; i < _propsInternalNull.length; i++) {
-            this[_propsInternalNull[i]] = null;
-        }
-        for (let i = 0; i < _propsInternalVec3.length; i++) {
-            this[_propsInternalVec3[i]] = new Float32Array(3);
-        }
-
-        this._chunks = new Chunks();
+        // set default values
+        Object.keys(_props).forEach((name) => {
```

Comment:
Use of forEach leads to function/scope allocations, once compiled to ES5.
It is way slower than using for loops, and for code that runs on each frame, forEach to be avoided.